src/tools/miri/src/bin/miri.rs

```diff
@@ -342,6 +342,7 @@ fn jemalloc_magic() {
     // linking, so we need to explicitly depend on the function.
     #[cfg(target_os = "macos")]
     {
+        #[allow(missing_unsafe_on_extern)]
```

Comment:
Instead of adding an enormous amount of `allow`, what about keeping the lint allow-by-default until the bootstrap compiler has `unsafe extern`, so that then we can start using it in the codebase? That avoids having to change everything twice...

Reply:
Yeah, totally. I guess that depends when do we want to land this change and then how much do we want to wait but yeah :).

Reply:
On old editions I think the RFC says it should be allow-by-default for now anyway?

Reply:
Ohh good catch. I've implemented Felix's proposal https://github.com/rust-lang/rfcs/pull/3484#issuecomment-1758275493 which I see now is out of sync.
Going to fix this accordingly when I have some time.